azerothcore/azerothcore-wotlk
# fix(core): Horde Warsong Gulch gates dont despawn

## Changes Proposed:
-  Add BG event to despawn gates, and amend the header to add enums for the event and the delay time

## Issues Addressed:

- Closes Issue 13375 https://github.com/azerothcore/azerothcore-wotlk/issues/13375
- Closes https://github.com/chromiecraft/chromiecraft/issues/3721

## SOURCE:

Video: https://youtu.be/GjjhNIMYd5Y?t=70

## Tests Performed:

- Does it build without errors?  Yes
- Did you test in-game? Yes, video posted (with 10 second timer, but PULL despawn timer changed to 5 seconds)
- Video with 10-second despawn delay: Video: https://youtu.be/7Mb2HUzojP4
- What did you test? Confirmed proper behavior
- On which OS did you test? Windows 10
- Describe any other tests performed: Ran through all the gates to ensure they are actually open

## How to Test the Changes:


1. Enter WSG battleground on Horde side
2. Wait for game to start and gates to fall down
3. Gate rubble should despawn after 5 seconds

## Known Issues and TODO List:


- Confirm behavior with many players in BG
- Confirm multiple BG simultaneously that gates don't despawn unexpectedly


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
